reactor-test/src/test/java/reactor/test/publisher/ColdTestPublisherTests.java

```diff
@@ -257,6 +257,41 @@ public void expectMinRequestedFailure() {
 		publisher.assertMinRequested(0);
 	}
 
+	@Test
+	public void expectMaxRequestedNormal() {
+		TestPublisher<String> publisher = TestPublisher.createCold();
+
+		Flux.from(publisher).limitRequest(5).subscribe();
+		publisher.assertMaxRequested(5);
+
+		Flux.from(publisher).limitRequest(10).subscribe();
+		publisher.assertSubscribers(2);
+		publisher.assertMaxRequested(10);
+	}
+
+	@Test
+	public void expectMaxRequestedWithUnbounded() {
+		TestPublisher<String> publisher = TestPublisher.createCold();
+
+		Flux.from(publisher).limitRequest(5).subscribe();
+		publisher.assertMaxRequested(5);
+
+		Flux.from(publisher).subscribe();
+		publisher.assertSubscribers(2);
+		publisher.assertMaxRequested(Long.MAX_VALUE);
+	}
+
+	@Test
+	public void expectMaxRequestedFailure() {
+		TestPublisher<String> publisher = TestPublisher.createCold();
+
+		Flux.from(publisher).limitRequest(5).subscribe();
+
+		assertThatExceptionOfType(AssertionError.class)
+				.isThrownBy(() -> publisher.assertMaxRequested(6))
+				.withMessage("Expected maximum request of 6; got 5");
```

Comment:
this "maximum" sounds like "6 or less", but actually it asserts that the requested amount of >= N - sounds like an inversion of wording

Reply:
What about "Expected the largest request to be at least N"?
It's interesting, actually -- now it's possible to verify that the smallest subscriber demand (through `assertMinRequested` and the largest subscriber demand (through `assertMaxRequested`) is at least some lower bound. But it's not possible to verify a demand upper bound on any subscriber.

Reply:
@elefeint do you mean that this doesn't solve your issue?

Reply:
note: both `assertMinRequested` and `assertMaxRequested` assert the _total_ **outstanding** requested amount from each subscriber. eg. `request(1)` followed by `request(4)` followed by testpublisher emitting 2 elements will but that number at `1 + 4 - 2` = `3`. Maybe this is not as straightforward as this (unfortunately overeagerly merged) change 😖

Reply:
aaaaand I see what you mean, the change is plain wrong: `maxRequest < n` instead of `maxRequest > n`, right?